clippy_lints/src/needless_pass_by_ref_mut.rs

```diff
@@ -244,8 +244,10 @@ impl<'tcx> LateLintPass<'tcx> for NeedlessPassByRefMut<'tcx> {
             .iter()
             .filter(|(def_id, _)| !self.used_fn_def_ids.contains(def_id))
         {
-            let show_semver_warning =
-                self.avoid_breaking_exported_api && cx.effective_visibilities.is_exported(*fn_def_id);
+            let is_exported = cx.effective_visibilities.is_exported(*fn_def_id);
+            if self.avoid_breaking_exported_api && is_exported {
+                continue;
+            }
```

Comment:
I think it would be cleaner to have this check in `check_fn` where the function is added to `NeedlessPassByRefMut::fn_def_ids_to_maybe_unused_mut`. But this is a viable position as well.

Reply:
https://github.com/rust-lang/rust-clippy/pull/11647/commits/125c778d6d01824fde3fe96f765a4dd9341af036